### DISCUSSION TITLE
Version 4 Add/Remove/Change

### DISCUSSION THREAD

**Discussion by pimbrouwers** [falcoframework/Falco#76] [Ideas]
Version 4 is actively being developed. What would you like to see added, removed or changed?

**Comment by adelarsq** [comment]
Is It possible something like graphql endpoints ([ref](https://fsprojects.github.io/FSharp.Data.GraphQL))?

```fsharp
...
let main args =
    webHost args {
        endpoints [ 
            gql query "people" ( blablabla )
            gql mutation "people" ( blablabla )
        ]
    }
    0
```

**Reply by pimbrouwers** [reply]
Maybe I'm wrong. But I think this would be best done as a separate package. The routing mechanism is just a simple abstraction on top of endpoint routing, so if that supports graphql then we can try and make this work in the main package.

**Reply by adelarsq** [reply]
Exactly. Just thinking a way that would be less verbose. At moment with POST endpoints it's already possible to support query and mutations.

**Reply by pimbrouwers** [reply]
Ok cool. Can you share snippet(s) of code you have already used to make this work?

**Reply by adelarsq** [reply]
@pimbrouwers Yes. I will take some time next weekend and create a sample project

**Reply by adelarsq** [reply]
@pimbrouwers Sorry, I forgot about this. So much work to do 🥵 
When it calms down a little I will take a look. At moment I use hotchocolate, but I think that can be easier.

**Comment by JacksonCribb** [comment]
Any thoughts on upgrading to the new WebApplicationBuilder & IHostBuilder? not a huge change but would sacrifice < 6.0 support. Not sure how important that is to Falco.

I have a deprecation warning from Serilog.AspNetcore `This construct is deprecated. Prefer UseSerilog() on IHostBuilder`. I'm sure others will start running into things that want the new builder too. 

The builder is generally pretty easy to extend & use, not sure it needs to be marketed as Falco centric; not a request, just have thoughts bubbling on what 'extensions' could look like I was actually just about to use it for a web host with health checks but not Falco features.

**Reply by pimbrouwers** [reply]
Happy to report this change has been made!
